test/config_test/example_configs_test.cc

```diff
@@ -21,9 +25,9 @@ TEST(ExampleConfigsTest, All) {
 
 #if defined(__APPLE__) || defined(WIN32)
   // freebind/freebind.yaml is not supported on macOS or Windows and is disabled via Bazel.
-  EXPECT_EQ(37UL, ConfigTest::run(directory));
+  EXPECT_EQ(NumberOfConfigurationFiles - 1, ConfigTest::run(directory));
```

Comment:
im thinking we should either move the `-1` to the script so we dont need this if clause, or we also set the windows/mac decrement somewhere

Reply:
+1

Reply:
Not sure if automating counting the `*.yaml` makes sense. 

```diff
diff --git a/test/config_test/BUILD b/test/config_test/BUILD
index 496cac0e82..792ba3d603 100644
--- a/test/config_test/BUILD
+++ b/test/config_test/BUILD
@@ -24,6 +24,7 @@ envoy_cc_test(
     ],
     deps = [
         ":config_test_lib",
+        "//source/common/filesystem:filesystem_lib",
         "//test/test_common:environment_lib",
         "//test/test_common:utility_lib",
     ],
diff --git a/test/config_test/example_configs_test.cc b/test/config_test/example_configs_test.cc
index 2e02d41608..b430638813 100644
--- a/test/config_test/example_configs_test.cc
+++ b/test/config_test/example_configs_test.cc
@@ -1,6 +1,7 @@
 #include "test/config_test/config_test.h"
 #include "test/test_common/environment.h"
 #include "test/test_common/utility.h"
+#include "common/filesystem/filesystem_impl.h"
 
 #include "gtest/gtest.h"
 
@@ -9,6 +10,15 @@ TEST(ExampleConfigsTest, All) {
   TestEnvironment::exec(
       {TestEnvironment::runfilesPath("test/config_test/example_configs_test_setup.sh")});
 
+#ifdef WIN32
+  Filesystem::InstanceImplWin32 file_system;
+#else
+  Filesystem::InstanceImplPosix file_system;
+#endif
+
+  const auto number_of_configs = std::stoi(
+      file_system.fileReadToEnd(TestEnvironment::temporaryDirectory() + "/number-of-configs.txt"));
+
   // Change working directory, otherwise we won't be able to read files using relative paths.
 #ifdef PATH_MAX
   char cwd[PATH_MAX];
@@ -21,9 +31,9 @@ TEST(ExampleConfigsTest, All) {
 
 #if defined(__APPLE__) || defined(WIN32)
   // freebind/freebind.yaml is not supported on macOS or Windows and is disabled via Bazel.
-  EXPECT_EQ(37UL, ConfigTest::run(directory));
+  EXPECT_EQ(number_of_configs - 1, ConfigTest::run(directory));
 #else
-  EXPECT_EQ(38UL, ConfigTest::run(directory));
+  EXPECT_EQ(number_of_configs, ConfigTest::run(directory));
 #endif
 
   ConfigTest::testMerge();
diff --git a/test/config_test/example_configs_test_setup.sh b/test/config_test/example_configs_test_setup.sh
index 876438e7be..b1478c60e0 100755
--- a/test/config_test/example_configs_test_setup.sh
+++ b/test/config_test/example_configs_test_setup.sh
@@ -5,3 +5,6 @@ set -e
 DIR="$TEST_TMPDIR"/test/config_test
 mkdir -p "$DIR"
 tar -xvf "$TEST_SRCDIR"/envoy/configs/example_configs.tar -C "$DIR"
+
+NUMBER_OF_CONFIGS=$(find "$DIR" -type f | grep .yaml | wc -l)
+echo "$NUMBER_OF_CONFIGS" > "$TEST_TMPDIR"/number-of-configs.txt
```

Reply:
> Not sure if automating counting the *.yaml makes sense.

hmm - trying to understand - my ~guess is that the expected count is to ensure it loops/tests the correct number of times - so counting the files is ok (as your patch suggests)

ultimately my goal in trying to automate this is so that you dont have to change anything when adding/removing a sandbox, and also no need to document that